utils/src/test/java/com/cloud/utils/TernaryTest.java

```diff
@@ -28,7 +28,7 @@ public void testEquals() {
         Assert.assertEquals(new Ternary<String, String, String>("a", "b", "c"), new Ternary<String, String, String>("a", "b", "c"));
         Assert.assertFalse(new Ternary<String, String, String>("a", "b", "c").equals(new Ternary<String, String, String>("a", "b", "d")));
         Assert.assertFalse(new Ternary<String, String, String>("a", "b", "c").equals(""));
-        Assert.assertFalse(new Ternary<String, String, String>("a", "b", "c").equals(null));
```

Comment:
You do not need to remove the `equals` here. In this very specific use case, there is no null pointer that can happen here.

Reply:
thanks for the hint, I did not know about this other specific use case.